litestar/_openapi/schema_generation/schema.py

```diff
@@ -519,7 +538,7 @@ def for_builtin_generics(self, field: SignatureField) -> Schema:
                 },
             )
 
-        cursor_schema = replace(self, generate_examples=False).for_field(field.children[0])  # type: ignore[index]
+        cursor_schema = self.not_generating_examples.for_field(field.children[0])  # type: ignore[index]
```

Comment:
so, this is a bit overcomplicated. 

Why not add `generate_examples` as a kwarg to for_field?

Reply:
Because it wouldn't be added just to `for_field` but to all (current and future) SchemaCreator methods, which is the whole point of this refactoring.